### PR TITLE
test: add more zcf tests (Part 2)

### PR DESCRIPTION
**Pull request by katelynsills**
This PR tests:

* zcf.assertUniqueKeyword
* zcf.saveIssuer
* zcf.makeInvitation

Still todo:

- [ ] reallocate
- [ ] shutdown
- [ ] makeZCFMint
- [ ] makeEmptySeatKit

Test ZCF objects like ZCFSeat.

Re: #1668

### CODE REVIEW

File: packages/zoe/test/unitTests/zcf/test-zcf.js [hunk 1]
```diff
@@ -194,3 +188,202 @@ test(`zcf.getAmountMath - from issuerKeywordRecord & zcf.saveIssuer`, async t =>
...
+  // Unique, but not a valid Keyword
+  t.throws(() => zcf.assertUniqueKeyword('a'), {
+    message:
+      'keyword "a" must be ascii and must start with a capital letter.\nSee console for error data.',
+  });
+  t.throws(() => zcf.assertUniqueKeyword('3'), {
+    message:
+      'keyword "3" must be ascii and must start with a capital letter.\nSee console for error data.',
+  });
+});
```
Comment (Chris-Hibbert) [inline packages/zoe/test/unitTests/zcf/test-zcf.js:212]: It's normal to include at least one positive test. Any reason not to here?

File: packages/zoe/test/unitTests/zcf/test-zcf.js [hunk 2]
```diff
@@ -194,3 +188,202 @@ test(`zcf.getAmountMath - from issuerKeywordRecord & zcf.saveIssuer`, async t =>
...
+    // TODO: improve error message
+    // https://github.com/Agoric/agoric-sdk/issues/1701
+    message: 'o["getBrand"] is not a function',
+  });
+});
+
+test(`zcf.saveIssuer - bad keyword`, async t => {
+  const { moolaKit } = setup();
+  const { zcf } = await setupZCFTest();
+  // TODO: why does this not throwAsync?
```
Comment (Chris-Hibbert) [inline packages/zoe/test/unitTests/zcf/test-zcf.js]: The assertion in saveIssuer comes before the async portion, so it throws immediately.

File: packages/zoe/test/unitTests/zcf/test-zcf.js [hunk 3]
```diff
@@ -194,3 +188,202 @@ test(`zcf.getAmountMath - from issuerKeywordRecord & zcf.saveIssuer`, async t =>
...
+  t.throws(() => zcf.makeInvitation(() => {}, { something: 'a' }), {
+    message: `invitations must have a description string: (an object)\nSee console for error data.`,
+  });
+});
+
+test(`zcf.makeInvitation - no customProperties`, async t => {
+  const { zcf, zoe, instance, installation } = await setupZCFTest();
+  const invitationP = zcf.makeInvitation(() => {}, 'myInvitation');
+  // TODO: allow promises for payments
+  // https://github.com/Agoric/agoric-sdk/issues/1705
```
Comment (Chris-Hibbert) [inline packages/zoe/test/unitTests/zcf/test-zcf.js:338]: The issue may have been revealed while writing the test, but it doesn't appear that fixing the issue will affect the test. If the test should be written differently once the issue is addressed, say so explicitly as on line 287.

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/zoe/test/unitTests/zcf/test-zcf.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/test/unitTests/zcf/test-zcf.js:212]: Yup, good point

File: packages/zoe/test/unitTests/zcf/test-zcf.js (hunk 2 above)
Comment (katelynsills) [reply packages/zoe/test/unitTests/zcf/test-zcf.js]: Ah, that seems like unpredictable behavior from a user's perspective. I'm going to change this to an async function such that it's always `throwsAsync` if it throws.

File: packages/zoe/test/unitTests/zcf/test-zcf.js (hunk 3 above)
Comment (katelynsills) [reply packages/zoe/test/unitTests/zcf/test-zcf.js:338]: It's only a types issue, so the test is not affected.

**Comment by katelynsills** [comment]
@Chris-Hibbert You already approved the test changes, but in fixing your suggestions on the tests, I changed the `saveIssuer` method and some of the types, if you want to take another look.

File: packages/zoe/src/internal-types.js [hunk 4]
```diff
@@ -200,7 +200,7 @@
 
 /**
  * @typedef {Object} IssuerTable
- * @propert {(brand: Brand) => boolean} hasByBrand
+ * @property {(brand: Brand) => boolean} hasByBrand
```
Comment (Chris-Hibbert) [inline packages/zoe/src/internal-types.js:203]: interesting!

File: packages/zoe/src/contractFacet/contractFacet.js [hunk 5]
```diff
@@ -265,18 +268,14 @@ export function buildRootObject() {
...
         zcf.assertUniqueKeyword(keyword);
-        return E(zoeInstanceAdmin)
-          .saveIssuer(issuerP, keyword)
-          .then(() => {
-            return issuerTable
-              .initIssuer(issuerP)
-              .then(record => registerIssuerRecordWithKeyword(keyword, record));
-          });
+        await E(zoeInstanceAdmin).saveIssuer(issuerP, keyword);
+        const record = await issuerTable.initIssuer(issuerP);
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractFacet/contractFacet.js]: I like the style that calls for comments highlighting the awaits.

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/zoe/src/contractFacet/contractFacet.js (hunk 5 above)
Comment (katelynsills) [reply packages/zoe/src/contractFacet/contractFacet.js]: Ok sounds good

**Pull request merged**